src/cata_utility.h

```diff
@@ -42,4 +42,48 @@ double logarithmic_range( int min, int max, int pos );
 int bound_mod_to_vals( int val, int mod, int max, int min );
 double convert_weight( int weight );
 
+/**
+* @brief Class used to access a list as if it were circular.
+*
+* Some times you just want to have a list loop around on itself.
+* This wrapper class allows you to do that. It requires the list to exist
+* separately, but that also means any changes to the list get propagated (both ways).
+*/
+template<typename T>
+class list_circularizer {
+    private:
+        unsigned int index = 0;
+        std::vector<T> *_list;
+    public:
+        list_circularizer( std::vector<T> &_list ) : _list( &_list )
```

Comment:
On all function headers? Or just on this one? ```CODE_STYLE.txt``` has them on the next line. (I personally prefer the style in your correction though).